packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -215,11 +216,24 @@ class PlatformAssetBundle extends CachingAssetBundle {
   @override
   Future<ByteData> load(String key) async {
     final Uint8List encoded = UTF8.encoder.convert(key);
-    final ByteData asset =
+    final ByteData assetPath =
         await BinaryMessages.send('flutter/assets', encoded.buffer.asByteData());
-    if (asset == null)
+
+    ByteData data;
+    if (assetPath != null) {
+      final File assetFile = new File(UTF8.decode(assetPath.buffer.asUint8List()));
+      final List<int> bytes = await assetFile.readAsBytes();
+      Uint8List uInt8list;
+      if (bytes is Uint8List) {
```

Comment:
just `assert(bytes is Uint8List)`. If it's ever NOT a Uint8List this is a serious performance concern.